specification/applicationinsights/resource-manager/microsoft.insights/stable/2015-05-01/componentWorkItemConfigs_API.json

```diff
@@ -0,0 +1,316 @@
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/microsoft.insights/components/{resourceName}/WorkItemConfigs": {
+      "get": {
+        "description": "Gets the list work item configurations that exist for the application",
+        "operationId": "WorkItemConfigurations_List",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "schema": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/WorkItemConfiguration"
+              }
+            }
+          },
+          "404": {
+            "description": "No configurations exist",
+            "schema": {
+              "$ref": "#/definitions/WorkItemConfigurationError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "WorkItemConfigurationsList": {
+            "$ref": "./examples/WorkItemConfigsGet.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      },
+      "post": {
+        "description": "Create a work item configuration for an Application Insights component.",
+        "operationId": "WorkItemConfigurations_Create",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "name": "WorkItemConfigurationProperties",
+            "description": "Properties that need to be specified to create a work item configuration of a Application Insights component.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/WorkItemCreateConfiguration"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "An object containing the work item configuration definition created by this PUT call.",
+            "schema": {
+              "$ref": "#/definitions/WorkItemConfiguration"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "WorkItemConfigurationsCreate": {
+            "$ref": "./examples/WorkItemConfigsCreate.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/microsoft.insights/components/{resourceName}/DefaultWorkItemConfig": {
```

Comment:
Why is the defaultWorkitemConfig part of this URL instead of a parameter as is the case in other APIs like DELETE, PUT etc.

Reply:
agreed, we'll fix this in our next release version.

Reply:
I dont understand. Is this just a swagger fix or a service side fix? I was hoping this is just an issue with how its described in swagger.

Reply:
This is the way the API was published. Essentially to retrieve default configuration for a resource. This is not a swagger fix, this is the swagger documentation for apis that are in PROD today. We're documenting our current public apis that are in production.